nylas/client/restful_models.py

```diff
@@ -874,6 +875,72 @@ def as_json(self):
         return dct
 
 
+class Webhook(NylasAPIObject):
+    attrs = [
+        "id",
+        "callback_url",
+        "state",
+        "triggers",
+        "application_id",
+        "version",
+    ]
+    read_only_attrs = {"id", "application_id", "version"}
+
+    collection_name = "webhooks"
+    api_root = "a"
+
+    def __init__(self, api):
+        NylasAPIObject.__init__(self, Webhook, api)
+
+    def as_json(self):
+        dct = {}
+        # Only 'state' can get updated
+        if self.id:
+            dct["state"] = self.state
+        else:
+            dct = NylasAPIObject.as_json(self)
+        return dct
+
+    class Trigger(str, Enum):
```

Comment:
I'd like some feedback on this, this is the first time I've introduced `Enum` to this SDK

Reply:
@mrashed-dev Should we call this Enum something like WebhookEnum? Just Enum sounds too generic

Reply:
Other than naming, the enum lgtm. People will be able to import it and compare it to strings. Just be aware that `Trigger.ACCOUNT_CONNECTED` will not return the string `"account.connected"`, you'll have to use `Trigger.ACCOUNT_CONNECTED.value`

Reply:
@tjovicic the name is `Trigger` not `Enum`, it just extends the `Enum` class. It's accessed by using `Webhook.Trigger.ACCOUNT_CONNECTED` for example.

Reply:
Oh, I didn't realize it was nested. I tend to only nest stuff if I really need to. But said that, it is a design choice, there is no real impact on the application by doing it this way, the enum will just be namespaced under the class name.